### PR TITLE
Add file path to displayName

### PR DESCRIPTION
**Pull request by aliaksandr-yermalayeu**
…ath and set it to displayName

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -149,11 +149,14 @@ function doesReturnJSX (body) {
   return false
 }
 
-function setDisplayNameAfter(path, nameNodeId, t, displayName) {
+function setDisplayNameAfter(state, path, nameNodeId, t, displayName) {
   if (!displayName) {
     displayName = nameNodeId.name
   }
 
+  var url = pathMod.relative(state.cwd, state.filename).replace(/\\/g, "/");
```
Comment (sergei-startsev) [inline index.js]: Could you rename `url` to `modulePath`?

File: index.js [hunk 2]
```diff
@@ -149,11 +149,14 @@ function doesReturnJSX (body) {
...
+  displayName = `${displayName} (${url})`;
```
Comment (sergei-startsev) [inline index.js]: maybe update `componentNameFromFilename` function and use it here? I don't see that this function is used anywhere...

**Comment by sergei-startsev** [comment]
Have you run tests? Could you also update them?

**Review by sergei-startsev — CHANGES_REQUESTED** [review]
Have you run tests? Could you also update them?

File: package.json [hunk 3]
```diff
@@ -3,13 +3,11 @@
   "version": "0.0.5",
```
Comment (asapach) [inline package.json:3]: you also need to update name and version

File: package.json [hunk 4]
```diff
@@ -18,12 +16,12 @@
   },
   "repository": {
     "type": "git",
-    "url": "git+https://github.com/opbeat/babel-plugin-add-react-displayname.git"
+    "url": "git+https://github.com/sergei-startsev/babel-plugin-react-displayname-path.git"
   },
   "author": "Ron Cohen",
```
Comment (asapach) [inline package.json:21]: and maybe do something with author? https://docs.npmjs.com/files/package.json#people-fields-author-contributors

File: test/fixtures/classComponents/expected.js [hunk 5]
```diff
@@ -1,36 +1,38 @@
-var _class, _temp;
+function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
```
Comment (asapach) [inline test/fixtures/classComponents/expected.js:1]: this is strange - where is it being used?

File: test/fixtures/decorators/expected.js [hunk 6]
```diff
@@ -1,13 +1,56 @@
...
+function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
 
 import React from 'react';
 import { Component } from 'react';
 import connect from '../decorators/connect';
 
-let DecoratedComponent = (_dec = connect(Component), _dec(_class = class DecoratedComponent extends React.Component {
-  render() {
-    return React.createElement('div', null);
+let DecoratedComponent = _decorate([connect(Component)], function (_initialize, _React$Component) {
```
Comment (asapach) [inline test/fixtures/decorators/expected.js:33]: omfg

is there an option to use something like https://babeljs.io/docs/en/babel-runtime?
or just delete this test

File: test/tests.js [hunk 7]
```diff
@@ -29,10 +25,11 @@ function readFile(filename) {
 
 function transformFile(filename) {
   return babel.transformFileSync(filename, {
-    presets: ['react', 'stage-1'],
+    presets: ['@babel/preset-react'],
     plugins: [
       [pluginPath, {'knownComponents': ['Component5a', 'Component5b', 'Component5c']}],
-      'transform-decorators-legacy',
+      ['@babel/plugin-proposal-decorators', {decoratorsBeforeExport: true}],
+      ['@babel/plugin-proposal-class-properties']
```
Comment (asapach) [inline test/tests.js:32]: is it actually being used anywhere?

**Review by asapach — CHANGES_REQUESTED** [review]

**Comment by asapach** [comment]
BTW, does this work with [memo](https://reactjs.org/docs/react-api.html#reactmemo)? I see people using it and the `displayName` is essentially useless. I think we need a test for this.

**Review by sergei-startsev — APPROVED** [review]

**Comment by sergei-startsev** [comment]
Merged PR to unlock further changes.

**Pull request merged**